src/ScrapingFunctions.re

```diff
@@ -29,9 +37,44 @@ let validateNonNull = x =>
 
 let toStringProcess = x => Js.String.make(x);
 
+/** return the date x days ago in the format yyyy-MM-dd */
+let daysAgoDate = (x: string) => {
+
+  /** formats a float date to a yyyy-MM-dd string */
+  let formatDate = date => Js.Date.fromFloat(date)
+    |> Js.Date.toISOString
+    |> {
+      /** ISO date format is yyyy-MM-ddThh:mm:ss:msms.nnnZ, we only need the first 10 characters*/
+      let startIsoOfset = 0;
+      let lengthIsoOfset = 10;
+      Js.String.substrAtMost(~from=startIsoOfset, ~length=lengthIsoOfset);
+    };
+
+  let now = Js.Date.make();
+  let delta = (Js.Date.getDate(now) -. float_of_int(int_of_string(x)));
+  Js.Float.isNaN(delta) ? "" : Js.Date.setDate(now, delta) |> formatDate;
+};
+
+let extractPostedDateProcess = x: string => {
+  let stringBody = x |> Js.String.make;
+
+  Js.String.match(postedDateRegex, stringBody)
```

Comment:
using lambdas with pipe operators is bad practice

instead: 

```
 let result = Js.String.match(postedDateRegex, stringBody);
  switch (result) {
  | None => ""
  | Some(match) => daysAgoDate(match[1])
  };
```

Reply:
thanks for the compliments :heart: 
and thanks for the best practices tip